cmd/goquery_completion/conditional.go

```diff
@@ -169,3 +224,151 @@ func conditional(args []string) []string {
+	if sugg.token == "|" && dirKeywordOccurrences > 0 && openParens == 0 {
+		return false
+	}
+
+	// if dir keyword has already occurred and there is already a top-level conjunction,
+	// no other 'unnested' conjunction or disjunction is allowed
+	//(otherwise dir keyword would not be part of the top-level conjunct anymore)
+	if sugg.token == "&" && dirKeywordOccurrences > 0 && topLevelAndOccurred && openParens == 0 {
+		return false
+	}
+	if (prev == "|" || prev == "&") && topLevelAndOccurred && dirKeywordOccurrences > 0 &&
+		firstUnnestedBinaryOpPos != nTokens-2 && openParens == 0 {
+		return false
+	}
+
+	return true
+}
+
+// conditionMustTerminate checks whether the condition string must end now in order to be a valid condition.
+// Returns true if condition string must be terminated and false otherwise.
+func conditionMustTerminate(prevprev, prev, last string, suggs []suggestion, openParens, dirKeywordOccurrences, nTokens int) bool {
+
+	// If there are no suggestions, and there is already a previous condition,
+	// and the current condition is a direction filter, the condition string
+	// must end (no other conditions are allowed afterwards).
+	// Note: this case handles a dir keyword occurring
+	// on the right of a top-level conjunction (potentially
+	// with trailing whitespaces)
+	// (e.g. terminates "sip = 127.0.0.1 & dir = in ",
+	// "sip = 127.0.0.1 & dir = inbound ")
+	if len(suggs) == 0 && nTokens > 3 {
+		for _, direction := range types.DirectionFilters {
+			// if dir is specified on the right side of a top-level conjunction,
+			// do not allow any trailing tokens
+			if prev == string(direction) && last == "" || last == string(direction) {
+				return true
+			}
+		}
+	}
+
+	// If there is exactly one suggestion, and the current
+	// condition is a direction filter and this suggestion is
+	// exactly what the user specified, the condition string must end.
+	// Note: this handles termination of condition strings ending
+	// with a direction filter in case there are trailing whitespaces.
+	// (e.g., terminates "sip = 127.0.0.1 & dir = inbound",
+	// "sip = 127.0.0.1 & dir = unidirectional" )
+	if len(suggs) == 1 && nTokens > 3 {
+		for _, direction := range types.DirectionFilters {
+			if suggs[0].token == string(direction) && suggs[0].token == last {
+				return true
+			}
+		}
+	}
+
+	// If there are no suggestions, and the dir keyword
+	// has already occurred, and the current position is
+	// directly after a top level condition with non-empty value,
+	// the condition string must end.
+	// The condition value is only validated for direction filters.
+	// Note: this case handles termination in case the
+	// dir keyword occurs on the left side of a conjunction
+	// (in the beginning of the condition string)
+	// (e.g. terminates "dir = in & sip = 127.0.0.1", BUT also
+	// "dir = in & sip = 42")
+	if len(suggs) == 0 && dirKeywordOccurrences > 0 && openParens == 0 && nTokens > 3 {
+
```

Comment:
remove empty leading line (wherever you open a block { )

Reply:
I'm with @basman here. Mainly because `golanci-lint` drilled this into my head at an early stage when I switched to VSCode.

Now that we have ourselves a little community, let us "formalize" some guidelines. Not that it must happen this PR, but in general, I propose:

* `golanci-lint` default settings as the linter base
* no unnamed returns. Even if return parameters are named in function definition
* no empty line between `{` and the next statements
* every log line starts the message _lowercase_
* comments: no preference as long as they don't collide with vet/lint. What I gravitate towards is more often than not a _lowercase_ letter at the start of an in-scope comment

Happy to put these down in writing once we converge on the above

Reply:
The guidelines seem very reasonable to me. 
I personally prefer an extra line in case the block starts with a comment as @fako1024 proposed, but if the default `golangci-lint` suggests to remove it, it's also fine with me.
With regards to comments my preference is to differentiate between single-sentence and multi-sentence comments. For single-sentence comments I prefer lowercase start without trailing dot, while for multi-sentence comments I prefer to stick with regular grammar rules. But I don't have a strong opinion on this.

Reply:
I'd lean towards @basman's opinion about comments following regular grammar (and, as stated before, leaving an empty line to the comment at the beginning of a block, which doesn't / shouldn't conflict with `golangci-lint`, I've been doing it all along), i.e. starting with an uppercase letter - always, even for a short comment (mostly for the reason of consistency / simplicity of rule, IMHO it unnecessarily complicates things if you have two different rules for virtually the same thing) - after all, a sentence is a sentence is a sentence. I'm not too pedantic about the dot though in the single sentence case...

As for the rest of the proposal I'm totally game (can't promise I won't forget a named return here and there though :see_no_evil:), thanks @els0r ...